src/ITIL_ValidationStep.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+abstract class ITIL_ValidationStep extends CommonDBTM
+{
+    public $dohistory              = false;
+
+    /**
+     * @var class-string<CommonITILValidation>
+     * To override in child classes
+     */
+    protected static string $validation_classname = TicketValidation::class;
+
+    public static function getTable($classname = null)
+    {
+        return $classname
+            ? parent::getTable($classname)
+            : parent::getTable(ITIL_ValidationStep::class);
+    }
+
+    public static function canCreate(): bool
+    {
+        return static::$validation_classname::canCreate();
+    }
+
+    /**
+     * Update right does not exist on ChangeValidation and TicketValidation, use create right
+     */
+    public static function canUpdate(): bool
+    {
+        return static::$validation_classname::canCreate();
+    }
+
+    /**
+     * View right does not exist on ChangeValidation and TicketValidation, use create right
+     */
+    public static function canView(): bool
+    {
+        return static::$validation_classname::canCreate();
+    }
```

Comment:
Should be handled already if this class extends CommonDBChild

Reply:
My knowledge on rights is limited, I can't answer. 
@cedric-anne can be removed ?

Reply:
This one is pretty specific. `ITIL_ValidationStep` is a child of a `CommonITILObject`, not from a `CommonITILValidation`, but the rights to create/update/delete a validation step should depend on the same condition as fot the validations.
For me, proxifying the methods like this is OK.